test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1

```diff
@@ -0,0 +1,129 @@
+$null = New-Item -ItemType Directory -Path $outputBaseFolder -Force
+
+$openCoverPath = "$outputBaseFolder\OpenCover"
+$testPath = "$outputBaseFolder\tests"
+$psBinPath = "$outputBaseFolder\PSCodeCoverage"
+$openCoverTargetDirectory = "$outputBaseFolder\OpenCoverToolset"
+$outputLog = "$outputBaseFolder\CodeCoverageOutput.xml"
+$psCodeZip = "$outputBaseFolder\PSCode.zip"
+$psCodePath = "$outputBaseFolder\PSCode"
+
+try
+{
+    Write-LogPassThru -Message "Starting downloads."
+
+    Invoke-WebRequest -uri $codeCoverageZip -outfile "$outputBaseFolder\PSCodeCoverage.zip"
+    Invoke-WebRequest $testContentZip -outfile "$outputBaseFolder\tests.zip"
+    Invoke-WebRequest $openCoverZip -outfile "$outputBaseFolder\OpenCover.zip"
+
+    Write-LogPassThru -Message "Downloads complete."
+
+    Write-LogPassThru -Message "Starting expansion."
+
+    Expand-Archive -path "$outputBaseFolder\PSCodeCoverage.zip" -destinationpath "$psBinPath" -Force
+    Expand-Archive -path "$outputBaseFolder\tests.zip" -destinationpath $testPath -Force
+    Expand-Archive -path "$outputBaseFolder\OpenCover.zip" -destinationpath $openCoverPath -Force
+
+    ## Download Coveralls.net uploader
+    $coverallsToolsUrl = 'https://github.com/csMACnz/coveralls.net/releases/download/0.7.0/coveralls.net.0.7.0.nupkg'
+    $coverallsPath = "$outputBaseFolder\coveralls"
+
+    ## Saving the nupkg as zip so we can expand it.
+    Invoke-WebRequest $coverallsToolsUrl -outfile "$outputBaseFolder\coveralls.zip"
+    Expand-Archive -Path "$outputBaseFolder\coveralls.zip" -DestinationPath $coverallsPath -Force
+
+    Write-LogPassThru -Message "Expansion complete."
+
+    Import-Module "$openCoverPath\OpenCover" -force
+    Install-OpenCover -TargetDirectory $openCoverTargetDirectory -force
+    Write-LogPassThru -Message "OpenCover installed."
+
+    Write-LogPassThru -Message "TestDirectory : $testPath"
+    Write-LogPassThru -Message "openCoverPath : $openCoverTargetDirectory\OpenCover"
+    Write-LogPassThru -Message "psbinpath : $psBinPath"
+
+    $openCoverParams = @{outputlog = $outputLog;
+                         TestDirectory = $testPath;
+                         OpenCoverPath = "$openCoverTargetDirectory\OpenCover";
+                         PowerShellExeDirectory = "$psBinPath\publish"
+                        }
+
+    $openCoverParams | Out-String | Write-LogPassThru
+    Write-LogPassThru -Message "Starting test run."
+
+    Invoke-OpenCover @openCoverParams
+
+    if(Test-Path $outputLog)
+    {
+        Write-LogPassThru -Message (get-childitem $outputLog).FullName
+    }
+
+    Write-LogPassThru -Message "Test run done."
+
+    $gitCommitId = & "$psBinPath\publish\powershell.exe" -noprofile -command { $PSVersiontable.GitCommitId }
+    $commitId = $gitCommitId.substring($gitCommitId.LastIndexOf('-g') + 2)
+
+    Write-LogPassThru -Message $commitId
+
+    $coverallsPath = "$outputBaseFolder\coveralls"
+    $coverallsToken = 'SaRLabTUDNqaO2JT9uuYcR78a7kia044D'
```

Comment:
Fixed.

---
In reply to: [94475097](https://github.com/PowerShell/PowerShell/pull/2918#discussion_r94475097) [](ancestors = 94475097)